### PR TITLE
Use cisagov/session-manager-tf-module to configure SSM Session Manager

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request makes use of [cisagov/session-manager-tf-module](https://github.com/cisagov/session-manager-tf-module) to set up access for and configure SSM Session Manager.

## 💭 Motivation and context ##

We want to:
- Stop using ssh tunneling over SSM
- Start logging SSM sessions
- DRY out some repeated code

## 🧪 Testing ##

These changes have been successfully applied to the staging COOL environment.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert dependencies to default branches.

### CODE REVIEW

File: shared-services/outputs.tf [hunk 1]
```diff
@@ -9,6 +9,6 @@ output "provisionaccount_role" {
 }
 
 output "ssmsession_role" {
-  value       = aws_iam_role.ssmsession_role
+  value       = module.session_manager.ssm_session_role
   description = "The IAM role that allows creation of SSM SessionManager sessions to any EC2 instance in this account."
```
Comment (dav3r) [inline shared-services/outputs.tf]: So this is where it all started.....
```suggestion
  description = "The IAM role that allows creation of SSM Session Manager sessions to any EC2 instance in this account."
```

**Review by dav3r — COMMENTED** [review]
This looks solid to me. 👍 👍  
Go forth and conquer the rest of the accounts in this repo. 🚀

File: shared-services/outputs.tf (hunk 1 above)
Comment (jsf9k) [reply shared-services/outputs.tf]: Please see commit 9264e5f.

**Comment by jsf9k** [comment]
> Go forth and conquer the rest of the accounts in this repo.

Given that we decided to only add this jazz to accounts that actually have EC2 instances running in them, no other accounts require these changes.  One could argue for adding them to the dynamic accounts, but given the aforementioned decision it makes more sense to add them _if_ the dynamic account will actually contain EC2 instances; for example, I will add these changes to the [cisagov/cool-assessment-terraform](https://github.com/cisagov/cool-assessment-terraform) repository.

**Review by dav3r — APPROVED** [review]
Fantastic! 💎

File: shared-services/provisionssmsessionmanager_policy_attachment.tf [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # ------------------------------------------------------------------------------
 # Attach to the ProvisionAccount role the IAM policy that allows
-# provisioning of the SSM Document resource required in this account.
+# provisioning of the SSM Document resource and set up SSM session
+# logging in this account.
```
Comment (mcdonnnj) [inline shared-services/provisionssmsessionmanager_policy_attachment.tf]: Just to align with the comment in `shared-services/provisionssmsessionmanager_policy.tf`. What drew my attention here was

```
# provisioning of the SSM Document resource and set up SSM session
```

so if you don't like this wholesale replacement I would at least like the `set` above changed.

```suggestion
# all of the permissions necessary to provision the SSM Document
# resource and set up SSM session logging in this account.
```

**Review by mcdonnnj — APPROVED** [review]
Nice improvement. I had one suggestion related to a comment.

File: shared-services/provisionssmsessionmanager_policy_attachment.tf (hunk 2 above)
Comment (jsf9k) [reply shared-services/provisionssmsessionmanager_policy_attachment.tf]: Please see commit 1bd322c.

**Pull request merged**